src/dotnet/APIView/APIViewWeb/Managers/CopilotCommentsManager.cs

```diff
@@ -31,163 +31,101 @@ IConfiguration configuration
+                document.BadCode = badCode;
+                document.Embedding = embedding;
             }
 
-            if (comment != null)
+            if (goodCode != null)
             {
-                update = update.Set("Comment", comment);
+                document.GoodCode = goodCode;
             }
 
-            if (guidelineIds != null)
+            if (comment != null)
             {
-                update = update.Set("GuidelineIds", guidelineIds);
+                document.Comment = comment;
             }
 
-            if (badCode != null)
+            if (guidelineIds != null)
             {
-                var embedding = await GetEmbeddingsAsync(badCode);
-                update = update
-                    .Set("BadCode", badCode)
-                    .Set("Embedding", embedding);
+                document.GuidelineIds = SplitGuidelineIds(guidelineIds);
             }
 
-            return await _copilotCommentsRepository.UpdateDocumentAsync(filter, update);
+            await _copilotCommentsRepository.UpdateDocumentAsync(document);
+            return GetDocumentJsonWithoutEmbedding(document);
         }
 
-        public Task DeleteDocumentAsync(string user, string id)
+        public Task DeleteDocumentAsync(string id, string language, string user)
         {
-            var filter = GetIdFilter(id);
-
-            var updateBuilder = Builders<CopilotCommentModel>.Update;
-            var update = updateBuilder
-                .Set("IsDeleted", true)
-                .Set("ModifiedOn", DateTime.UtcNow)
-                .Set("ModifiedBy", user);
-
-            return _copilotCommentsRepository.DeleteDocumentAsync(filter, update); 
+            return _copilotCommentsRepository.DeleteDocumentAsync(id, language, user); 
         }
 
-        public async Task<string> GetDocumentAsync(string id)
+        public async Task<string> GetDocumentAsync(string id, string language)
         {
-            var filter = GetIdFilter(id);
-            var document = await _copilotCommentsRepository.GetDocumentAsync(filter);
-
-            var documentJson = JsonConvert.SerializeObject(document);
-
-            return documentJson;
+            var document = await _copilotCommentsRepository.GetDocumentAsync(id, language);
+            return GetDocumentJsonWithoutEmbedding(document);
         }
 
-        private FilterDefinition<CopilotCommentModel> GetIdFilter(string id)
+        public async Task<string> SearchDocumentsAsync(string language, string badCode, float threshold, int limit)
         {
-            return Builders<CopilotCommentModel>.Filter
-                .Where(f => f.Id.ToString() == id && f.IsDeleted == false);
+            var embeddings = await GetEmbeddingsAsync(badCode);
+            var searchResults = await _copilotCommentsRepository.SimilaritySearchAsync(language, embeddings, threshold, limit);
+            var topResults = searchResults.Where(x => x.similarity >= threshold).AsEnumerable();
```

Comment:
Minor nit - Do we need the `.AsEnumerable()` at the end?  [`.Where(...)`](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.where?view=net-7.0#system-linq-enumerable-where-1(system-collections-generic-ienumerable((-0))-system-func((-0-system-boolean)))) also returns `IEnumerable<T>`.